### DISCUSSION TITLE
打包后的umi.js中的 数字类型的属性能通过配置修改成字符数字吗

### DISCUSSION THREAD

**Discussion by majiorxiu** [umijs/umi#12080] [Q&A] [answered]
![image](https://github.com/umijs/umi/assets/12382161/317ba39e-0fc1-4b00-ac98-23466d422825)
如题
起因是因为我使用微前端框架 JD的micro-app，本地与测试环境都可以正常访问，但在正式环境上，基座访问子应用是 由子应用的umi.js报出如下错误
umi.c9ecbbff.js:625 [micro-app from runScript] app iot:  TypeError: Cannot read properties of undefined (reading 'call')
    at Ne (umi.66131225.js:583:2817)

**Comment by fz6m** [comment, marked as answer]
umi 推荐用 qiankun 做微前端，并且官方也内置提供了更简便易用的 qiankun 插件。

如果你要用其他的微前端框架，要确保符合那个框架的前置要求（比如修改一些 webpack 配置），如有问题，请自行排查，或反馈到对应框架的 issue 里。

关于压缩成 key 是 number 这种情况，这是符合 js 规范的，没有任何问题，如果不兼容这种情况（我不确定你遇到的问题到底是不是因为 key 为 number 导致的），需要微前端框架来兼容，因为压缩后的结果只要符合 js 规范能运行，那这份产物就是正确的。

可以通过修改 [`jsMinifier`](https://umijs.org/docs/api/config#jsminifier-webpack) 选项来切换使用其他的代码压缩器，但我认为这无济于事，因为 ts 的匿名枚举和第三方库是否使用了匿名枚举，或使用了 key 为 number 的写法，你并不能确定，所以从 key 的角度来解决问题方向是错误的。
